king_phisher/client/dialogs/configuration.py

```diff
@@ -274,6 +279,31 @@ def interact(self):
 		self.dialog.destroy()
 		return response
 
+	def save_proxy_settings(self):
+		proxy_url = urllib.parse.urlparse(self.gtk_builder_get('entry_proxy_username').get_text())
+		proxy_username = self.gtk_builder_get('entry_proxy_username').get_text()
+		proxy_password = self.gtk_builder_get('entry_proxy_password').get_text()
+		if not proxy_url.hostname and not proxy_url.scheme:
+			gui_utilities.show_dialog_warning('Invalid Proxy Values', self.parent,
+												'The proxy settings you have entered are not valid.')
+			return
+		if proxy_username and proxy_password and proxy_url.port:
```

Comment:
You can convert the `ParseResult` returned by `urllib.parse.urlparse` with `urllib.parse.urlunparse` instead of using a format string.